OPAL/tac/src/main/scala/org/opalj/tac/fpcf/analyses/ifds/IFDSBasedVariableTypeAnalysis.scala

```diff
@@ -0,0 +1,106 @@
+/* BSD 2-Clause License - see OPAL/LICENSE for details. */
+package org.opalj.tac.fpcf.analyses.ifds
+
+import org.opalj.br.analyses.SomeProject
+import org.opalj.fpcf.{PropertyBounds, PropertyKey, PropertyStore}
+import org.opalj.ifds.{IFDSProperty, IFDSPropertyMetaInformation}
+
+import org.opalj.tac.fpcf.properties.TACAI
+import org.opalj.tac.fpcf.properties.cg.Callers
+import java.io.{File, PrintWriter}
+
+import org.opalj.ifds.IFDSAnalysis
+import org.opalj.ifds.IFDSAnalysisScheduler
+import org.opalj.ifds.Statistics
+
+import org.opalj.br.Method
+import org.opalj.br.analyses.DeclaredMethodsKey
+import org.opalj.br.analyses.ProjectInformationKeys
+import org.opalj.br.fpcf.PropertyStoreKey
+import org.opalj.tac.cg.TypeIteratorKey
+
+/**
+ * A variable type analysis implemented as an IFDS analysis.
+ * In contrast to an ordinary variable type analysis, which also determines types of fields,
+ * this analysis only determines the types of local variables.
+ * The subsuming taint can be mixed in to enable subsuming.
+ *
+ * @param project The analyzed project.
+ * @author Mario Trageser
+ */
+class IFDSBasedVariableTypeAnalysis(project: SomeProject, subsumeFacts: Boolean = false)
```

Comment:
Should maybe move individual analyses like this to a different package than the base infrastructure.